packages/block-library/src/gallery/edit.js

```diff
@@ -217,9 +217,19 @@ function GalleryEdit( props ) {
 	}
 
 	function isValidFileType( file ) {
+		// It's necessary to retrieve the media type from the raw image data for already-uploaded images on native.
+		const nativeBlockData =
```

Comment:
This is not a blocker but I was thinking if we could rename this variable to something like `nativeFileData` since it contains the current file data. What do you think?